Unable to run Android app on stable  2.10 if it uses a plugin with an android folder but no Android support

## Steps to Reproduce

The app can't be built on Android if a plugin is only iOS targetting. 
It's the case with [apple_map_flutter](https://pub.dev/packages/apple_maps_flutter). 

I can't run my app under Android with the new release ```Flutter: 2.10``` because the tool expect all plugins to support both Android and iOS or something like this (i have another project that builds find without `apple_map_flutter`)

1. Create a new project under flutter: 2.10
2. Add [apple_maps_flutter](https://pub.dev/packages/apple_maps_flutter) dependency using `flutter pub add apple_maps_flutter`
3. Build your app on Android device on flutter 2.10 stable/master

**Expected results:** <!-- what did you want to see? -->

The app should be built on Android (and on iOS) even if a plugin is missing for one of the platform, as it was done before ```flutter: 2.10```.

**Actual results:** <!-- what did you see? -->

```
Launching lib/main.dart on moto g 8 power in debug mode...
Running Gradle task 'assembleDebug'...
Plugin project :apple_maps_flutter not found. Please update settings.gradle.

FAILURE: Build failed with an exception.

* Where:
Script '/Users/foxtom/Desktop/flutter/packages/flutter_tools/gradle/flutter.gradle' line: 421

* What went wrong:
A problem occurred evaluating root project 'android'.
> A problem occurred configuring project ':app'.
   > Cannot invoke method afterEvaluate() on null object
```

<details>
  <summary>Logs</summary>

<!-- Finally, paste the output of running `flutter doctor -v` here. -->

```
[✓] Flutter (Channel stable, 2.10.0, on macOS 12.2 21D49 darwin-x64, locale fr-FR)
    • Flutter version 2.10.0 at /Users/foxtom/Desktop/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 5f105a6ca7 (2 days ago), 2022-02-01 14:15:42 -0800
    • Engine revision 776efd2034
    • Dart version 2.16.0
    • DevTools version 2.9.2

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/foxtom/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7590822)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7590822)

[✓] VS Code (version 1.63.2)
    • VS Code at /Users/foxtom/Desktop/Visual Studio Code.app/Contents
    • Flutter extension version 3.34.0

[✓] Connected device (3 available)
    • moto g 8 power (mobile) • ZY22BNDW2C • android-arm64  • Android 11 (API 30)
    • macOS (desktop)         • macos      • darwin-x64     • macOS 12.2 21D49 darwin-x64
    • Chrome (web)            • chrome     • web-javascript • Google Chrome 98.0.4758.80

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

</details>

Hi @Tom3652, Thanks for filing the issue. I  was able to reproduce the same behavior the app failed to build when using a plugin not supported for Android. This was reproducible on the latest stable/master and even on `stable 2.8.1`. This looks like a duplicate of https://github.com/flutter/flutter/issues/46304 (closed). Not sure if this is intended, But I was able to fix this by modifying the `settings.gradle`

<details>
<summary>settings.gradle (after modification)</summary>

```gradle
include ':app'

def localPropertiesFile = new File(rootProject.projectDir, "local.properties")
def properties = new Properties()

assert localPropertiesFile.exists()
localPropertiesFile.withReader("UTF-8") { reader -> properties.load(reader) }

def flutterSdkPath = properties.getProperty("flutter.sdk")
assert flutterSdkPath != null, "flutter.sdk not set in local.properties"
apply from: "$flutterSdkPath/packages/flutter_tools/gradle/app_plugin_loader.gradle"

def flutterProjectRoot = rootProject.projectDir.parentFile.toPath()

// Fixed by adding the lines below

def plugins = new Properties()
def pluginsFile = new File(flutterProjectRoot.toFile(), '.flutter-plugins')
if (pluginsFile.exists()) {
    pluginsFile.withReader('UTF-8') { reader -> plugins.load(reader) }
}

plugins.each { name, path ->
    def pluginDirectory = flutterProjectRoot.resolve(path).resolve('android').toFile()
    include ":$name"
    project(":$name").projectDir = pluginDirectory
}
```
</details>


<details>
<summary>flutter run stable(2.10)</summary>

```
Launching lib[/main.dart]() on Redmi K20 Pro in debug mode...
lib/main.dart:1
Plugin project :apple_maps_flutter not found. Please update settings.gradle.

FAILURE: Build failed with an exception.

* Where:
Script '[/Users/mahesh/Documents/flutter/packages/flutter_tools/gradle/flutter.gradle]()' line: 421

* What went wrong:
A problem occurred evaluating root project 'android'.
> A problem occurred configuring project ':app'.
   > Cannot invoke method afterEvaluate() on null object

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at [https://help.gradle.org]()

BUILD FAILED in 21s
Exception: Gradle task assembleDebug failed with exit code 1
Exited (sigterm)
```
</details>

<details>
<summary>flutter run (stable 2.8.1)</summary>

```console
mahesh@Maheshs-MacBook-Air-M1 sample % flutter run     
Launching lib/main.dart on Redmi K20 Pro in debug mode...
Plugin project :apple_maps_flutter not found. Please update settings.gradle.
Warning: Mapping new ns http://schemas.android.com/repository/android/common/02 to old ns http://schemas.android.com/repository/android/common/01
Warning: Mapping new ns http://schemas.android.com/repository/android/generic/02 to old ns http://schemas.android.com/repository/android/generic/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/addon2/02 to old ns http://schemas.android.com/sdk/android/repo/addon2/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/repository2/02 to old ns http://schemas.android.com/sdk/android/repo/repository2/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/sys-img2/02 to old ns http://schemas.android.com/sdk/android/repo/sys-img2/01
Warning: Mapping new ns http://schemas.android.com/repository/android/common/02 to old ns http://schemas.android.com/repository/android/common/01
Warning: Mapping new ns http://schemas.android.com/repository/android/generic/02 to old ns http://schemas.android.com/repository/android/generic/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/addon2/02 to old ns http://schemas.android.com/sdk/android/repo/addon2/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/repository2/02 to old ns http://schemas.android.com/sdk/android/repo/repository2/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/sys-img2/02 to old ns http://schemas.android.com/sdk/android/repo/sys-img2/01
e: /Users/mahesh/Desktop/sample/android/app/src/main/kotlin/com/example/sample/MainActivity.kt: (3, 19): Unresolved reference: embedding
e: /Users/mahesh/Desktop/sample/android/app/src/main/kotlin/com/example/sample/MainActivity.kt: (5, 21): Unresolved reference: FlutterActivity

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:compileDebugKotlin'.
> Compilation error. See log for more details

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 35s
Running Gradle task 'assembleDebug'...                             37.2s
Exception: Gradle task assembleDebug failed with exit code 1
```
</details>



<details>
<summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel stable, 2.10.0, on macOS 12.1 21C52 darwin-arm, locale en-GB)
    • Flutter version 2.10.0 at /Users/mahesh/Documents/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 5f105a6ca7 (2 days ago), 2022-02-01 14:15:42 -0800
    • Engine revision 776efd2034
    • Dart version 2.16.0
    • DevTools version 2.9.2

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/mahesh/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • ANDROID_HOME = /Users/mahesh/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.10.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)

[✓] IntelliJ IDEA Community Edition (version 2021.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 60.1.4
    • Dart plugin version 212.5080.8

[✓] VS Code (version 1.63.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.34.0

[✓] Connected device (3 available)
    • Redmi K20 Pro (mobile) • 192.168.1.2:5555 • android-arm64  • Android 11 (API 30)
    • macOS (desktop)        • macos            • darwin-arm64   • macOS 12.1 21C52 darwin-arm
    • Chrome (web)           • chrome           • web-javascript • Google Chrome 98.0.4758.80

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

```
[✓] Flutter (Channel master, 2.10.0-1.0.pre.371, on macOS 12.1 21C52 darwin-arm, locale en-GB)
    • Flutter version 2.10.0-1.0.pre.371 at /Users/mahesh/Documents/flutter_master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision d1285eef1b (2 days ago), 2022-02-02 00:50:20 -0500
    • Engine revision 5bbdb7a6f7
    • Dart version 2.17.0 (build 2.17.0-74.0.dev)
    • DevTools version 2.10.0

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/mahesh/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • ANDROID_HOME = /Users/mahesh/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.10.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)

[✓] IntelliJ IDEA Community Edition (version 2021.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 60.1.4
    • Dart plugin version 212.5080.8

[✓] VS Code (version 1.63.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.34.0

[✓] Connected device (3 available)
    • Redmi K20 Pro (mobile) • 192.168.1.2:5555 • android-arm64  • Android 11 (API 30)
    • macOS (desktop)        • macos            • darwin-arm64   • macOS 12.1 21C52 darwin-arm
    • Chrome (web)           • chrome           • web-javascript • Google Chrome 98.0.4758.80

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```
</details>

 cc: @blasten

--- a/packages/flutter_tools/gradle/app_plugin_loader.gradle
+++ b/packages/flutter_tools/gradle/app_plugin_loader.gradle
@@ -20,6 +20,7 @@ assert object instanceof Map
 assert object.plugins instanceof Map
 assert object.plugins.android instanceof List
 // Includes the Flutter plugins that support the Android platform.
+// This logic must be kept in sync with the logic in flutter.gradle.
 object.plugins.android.each { androidPlugin ->
   assert androidPlugin.name instanceof String
   assert androidPlugin.path instanceof String

--- a/packages/flutter_tools/gradle/flutter.gradle
+++ b/packages/flutter_tools/gradle/flutter.gradle
@@ -515,7 +515,7 @@ class FlutterPlugin implements Plugin<Project> {
     /** Gets the list of plugins that support the Android platform. */
     private Properties getPluginList() {
         Map meta = getDependenciesMetadata()
-        assert meta != null : "Please run `flutter pub get` and run this command again"
+        assert meta != null : 'Please run `flutter pub get` and run this command again'
         assert meta.plugins instanceof Map
         assert meta.plugins.android instanceof List
         Properties androidPlugins = new Properties()
@@ -559,7 +559,7 @@ class FlutterPlugin implements Plugin<Project> {
         // `plugin-b` depends on `plugin-c`.
         // `plugin-c` doesn't depend on anything.
         Map meta = getDependenciesMetadata()
-        assert meta != null : "Please run `flutter pub get` and call this command again"
+        assert meta != null : 'Please run `flutter pub get` and call this command again'
         assert meta.dependencyGraph instanceof List
         return meta.dependencyGraph
     }

--- a/packages/flutter_tools/gradle/flutter.gradle
+++ b/packages/flutter_tools/gradle/flutter.gradle
@@ -515,10 +515,12 @@ class FlutterPlugin implements Plugin<Project> {
     /** Gets the list of plugins that support the Android platform. */
     private Properties getPluginList() {
         Map meta = getDependenciesMetadata()
-        assert meta != null : 'Please run `flutter pub get` and run this command again'
+        Properties androidPlugins = new Properties()
+        if (meta == null) {
+            return androidPlugins
+        }
         assert meta.plugins instanceof Map
         assert meta.plugins.android instanceof List
-        Properties androidPlugins = new Properties()
 
         // This logic must be kept in sync with the logic in app_plugin_loader.gradle.
         meta.plugins.android.each { androidPlugin ->
@@ -559,7 +561,9 @@ class FlutterPlugin implements Plugin<Project> {
         // `plugin-b` depends on `plugin-c`.
         // `plugin-c` doesn't depend on anything.
         Map meta = getDependenciesMetadata()
-        assert meta != null : 'Please run `flutter pub get` and call this command again'
+        if (meta == null) {
+            return []
+        }
         assert meta.dependencyGraph instanceof List
         return meta.dependencyGraph
     }

--- a/packages/flutter_tools/gradle/flutter.gradle
+++ b/packages/flutter_tools/gradle/flutter.gradle
@@ -459,7 +459,7 @@ class FlutterPlugin implements Plugin<Project> {
 
             getPluginList().each { plugin ->
                 Project pluginProject = project.rootProject.findProject(plugin.key)
-                if (!pluginProject) {
+                if (pluginProject == null) {
                     return
                 }
                 pluginProject.afterEvaluate {

--- a/packages/flutter_tools/gradle/flutter.gradle
+++ b/packages/flutter_tools/gradle/flutter.gradle
@@ -48,7 +48,7 @@ class FlutterExtension {
      * Specifies the relative directory to the Flutter project directory.
      * In an app project, this is ../.. since the app's build.gradle is under android/app.
      */
-    String source
+    String source = '../..'
 
     /** Allows to override the target file. Otherwise, the target is lib/main.dart. */
     String target
@@ -418,7 +418,7 @@ class FlutterPlugin implements Plugin<Project> {
 
     /**
      * Compares semantic versions ignoring labels.
-     * 
+     *
      * If the versions are equal (ignoring labels), returns one of the two strings arbitrarily.
      *
      * If minor or patch are omitted (non-conformant to semantic versioning), they are considered zero.
@@ -459,6 +459,9 @@ class FlutterPlugin implements Plugin<Project> {
 
             getPluginList().each { plugin ->
                 Project pluginProject = project.rootProject.findProject(plugin.key)
+                if (!pluginProject) {
+                    return
+                }
                 pluginProject.afterEvaluate {
                     int pluginCompileSdkVersion = pluginProject.android.compileSdkVersion.substring(8) as int
                     maxPluginCompileSdkVersion = Math.max(pluginCompileSdkVersion, maxPluginCompileSdkVersion)
@@ -476,15 +479,7 @@ class FlutterPlugin implements Plugin<Project> {
                     }
                 }
             }
-        }  
-    }
-
-    /**
-     * Returns `true` if the given path contains an `android/build.gradle` file.
-     */
-    private Boolean doesSupportAndroidPlatform(String path) {
-        File editableAndroidProject = new File(path, 'android' + File.separator + 'build.gradle')
-        return editableAndroidProject.exists()
+        }
     }
 
     /**
@@ -495,8 +490,7 @@ class FlutterPlugin implements Plugin<Project> {
     private void configurePluginDependencies(Object dependencyObject) {
         assert dependencyObject.name instanceof String
         Project pluginProject = project.rootProject.findProject(":${dependencyObject.name}")
-        if (pluginProject == null ||
-            !doesSupportAndroidPlatform(pluginProject.projectDir.parentFile.path)) {
+        if (pluginProject == null) {
             return
         }
         assert dependencyObject.dependencies instanceof List
@@ -506,8 +500,7 @@ class FlutterPlugin implements Plugin<Project> {
                 return
             }
             Project dependencyProject = project.rootProject.findProject(":$pluginDependencyName")
-            if (dependencyProject == null ||
-                !doesSupportAndroidPlatform(dependencyProject.projectDir.parentFile.path)) {
+            if (dependencyProject == null) {
                 return
             }
             // Wait for the Android plugin to load and add the dependency to the plugin project.
@@ -519,17 +512,25 @@ class FlutterPlugin implements Plugin<Project> {
         }
     }
 
+    /** Gets the list of plugins that support the Android platform. */
     private Properties getPluginList() {
-        File pluginsFile = new File(project.projectDir.parentFile.parentFile, '.flutter-plugins')
-        Properties allPlugins = readPropertiesIfExist(pluginsFile)
+        Map meta = getDependenciesMetadata()
+        assert meta != null : "Please run `flutter pub get` and run this command again"
+        assert meta.plugins instanceof Map
+        assert meta.plugins.android instanceof List
         Properties androidPlugins = new Properties()
-        allPlugins.each { name, path ->
-            if (doesSupportAndroidPlatform(path)) {
-                androidPlugins.setProperty(name, path)
+
+        // This logic must be kept in sync with the logic in app_plugin_loader.gradle.
+        meta.plugins.android.each { androidPlugin ->
+            assert androidPlugin.name instanceof String
+            assert androidPlugin.path instanceof String
+            // Skip plugins that have no native build (such as a Dart-only implementation
+            // of a federated plugin).
+            def needsBuild = androidPlugin.containsKey('native_build') ? androidPlugin['native_build'] : true
+            if (!needsBuild) {
+                return
             }
-            // TODO(amirh): log an error if this plugin was specified to be an Android
-            // plugin according to the new schema, and was missing a build.gradle file.
-            // https://github.com/flutter/flutter/issues/40784
+            androidPlugins.setProperty(androidPlugin.name, androidPlugin.path)
         }
         return androidPlugins
     }
@@ -557,14 +558,29 @@ class FlutterPlugin implements Plugin<Project> {
         // This means, `plugin-a` depends on `plugin-b` and `plugin-c`.
         // `plugin-b` depends on `plugin-c`.
         // `plugin-c` doesn't depend on anything.
-        File pluginsDependencyFile = new File(project.projectDir.parentFile.parentFile, '.flutter-plugins-dependencies')
+        Map meta = getDependenciesMetadata()
+        assert meta != null : "Please run `flutter pub get` and call this command again"
+        assert meta.dependencyGraph instanceof List
+        return meta.dependencyGraph
+    }
+
+    private Map parsedFlutterPluginsDependencies
+
+    /**
+     * Parses <project-src>/.flutter-plugins-dependencies
+     */
+    private Map getDependenciesMetadata() {
+        if (parsedFlutterPluginsDependencies) {
+            return parsedFlutterPluginsDependencies
+        }
+        File pluginsDependencyFile = new File(getFlutterSourceDirectory(), '.flutter-plugins-dependencies')
         if (pluginsDependencyFile.exists()) {
             def object = new JsonSlurper().parseText(pluginsDependencyFile.text)
             assert object instanceof Map
-            assert object.dependencyGraph instanceof List
-            return object.dependencyGraph
+            parsedFlutterPluginsDependencies = object
+            return object
         }
-        return []
+        return null
     }
 
     private static String toCammelCase(List<String> parts) {

--- a/packages/flutter_tools/test/integration.shard/android_skip_unsupported_plugin.dart
+++ b/packages/flutter_tools/test/integration.shard/android_skip_unsupported_plugin.dart
@@ -1,0 +1,120 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:file_testing/file_testing.dart';
+import 'package:flutter_tools/src/base/file_system.dart';
+import 'package:flutter_tools/src/base/io.dart';
+import 'package:flutter_tools/src/cache.dart';
+
+import '../src/common.dart';
+import 'test_utils.dart';
+
+void main() {
+  late Directory tempDir;
+
+  setUp(() {
+    Cache.flutterRoot = getFlutterRoot();
+    tempDir = createResolvedTempDirectorySync('flutter_plugin_test.');
+  });
+
+  tearDown(() async {
+    tryToDelete(tempDir);
+  });
+
+  // Regression test for https://github.com/flutter/flutter/issues/97729.
+  test('skip plugin if it does not support the Android platform', () async {
+    final String flutterBin = fileSystem.path.join(
+      getFlutterRoot(),
+      'bin',
+      'flutter',
+    );
+
+    // Create dummy plugin that *only* supports iOS.
+    processManager.runSync(<String>[
+      flutterBin,
+      ...getLocalEngineArguments(),
+      'create',
+      '--template=plugin',
+      '--platforms=ios',
+      'test_plugin',
+    ], workingDirectory: tempDir.path);
+
+    final Directory pluginAppDir = tempDir.childDirectory('test_plugin');
+
+    // Create an android directory and a build.gradle file within.
+    final File pluginGradleFile = pluginAppDir
+        .childDirectory('android')
+        .childFile('build.gradle')
+      ..createSync(recursive: true);
+    expect(pluginGradleFile, exists);
+
+    pluginGradleFile.writeAsStringSync(r'''
+buildscript {
+    ext.kotlin_version = '1.5.31'
+    repositories {
+        google()
+        mavenCentral()
+    }
+
+    dependencies {
+        classpath 'com.android.tools.build:gradle:7.0.0'
+        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
+    }
+
+    configurations.classpath {
+        resolutionStrategy.activateDependencyLocking()
+    }
+}
+
+allprojects {
+    repositories {
+        google()
+        mavenCentral()
+    }
+}
+
+rootProject.buildDir = '../build'
+
+subprojects {
+    project.buildDir = "${rootProject.buildDir}/${project.name}"
+}
+
+subprojects {
+    project.evaluationDependsOn(':app')
+}
+
+task clean(type: Delete) {
+    delete rootProject.buildDir
+}
+''');
+
+    final Directory pluginExampleAppDir =
+        pluginAppDir.childDirectory('example');
+
+    // Run flutter build apk to build plugin example project.
+    final ProcessResult addAndroidResult = processManager.runSync(<String>[
+      flutterBin,
+      ...getLocalEngineArguments(),
+      'create',
+      '--template=app',
+      '--platforms=android',
+      '.',
+    ], workingDirectory: pluginExampleAppDir.path);
+    expect(addAndroidResult.exitCode, equals(0),
+        reason:
+            'flutter create exited with non 0 code: ${addAndroidResult.stderr}');
+
+    // Run flutter build apk to build plugin example project.
+    final ProcessResult buildApkResult = processManager.runSync(<String>[
+      flutterBin,
+      ...getLocalEngineArguments(),
+      'build',
+      'apk',
+      '--debug',
+    ], workingDirectory: pluginExampleAppDir.path);
+    expect(buildApkResult.exitCode, equals(0),
+        reason:
+            'flutter build apk exited with non 0 code: ${buildApkResult.stderr}');
+  });
+}

--- a/packages/flutter_tools/test/integration.shard/android_skip_unsupported_plugin.dart
+++ b/packages/flutter_tools/test/integration.shard/android_skip_unsupported_plugin.dart
@@ -92,7 +92,7 @@ task clean(type: Delete) {
     final Directory pluginExampleAppDir =
         pluginAppDir.childDirectory('example');
 
-    // Run flutter build apk to build plugin example project.
+    // Add android support to the plugin's example app.
     final ProcessResult addAndroidResult = processManager.runSync(<String>[
       flutterBin,
       ...getLocalEngineArguments(),

